### PR TITLE
[PUB-1972] Add segment tracking preview

### PR DESCRIPTION
**Pull request by anafilipadealmeida**
## Description
Add Segment tracking to Story Preview Opened (through Composer and Queue)

## Context & Notes
[PUB-1972](https://buffer.atlassian.net/browse/PUB-1972)

## Checklist

-   [x] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have tested different user stories. 
-   [x] I have identified similar or related features and tested they are still working.
-   [x] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [x] I have commented my code, particularly in hard-to-understand areas.

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/stories/index.js [hunk 1]
```diff
@@ -61,6 +63,24 @@ export default connect(
...
+      const ctaProperties = getCtaProperties(SEGMENT_NAMES.STORIES_PREVIEW_QUEUE);
+      const imageCount = stories.filter(story => story.type === 'image').length;
+      const videoCount = stories.filter(story => story.type === 'video').length;
+      const noteCount = stories.filter(story => story.note && story.note.length > 0).length;
+      const metadata = {
+        product: 'publish',
+        storyGroupId: id,
+        channel: 'instagram',
+        channelId: profileId,
+        channelServiceId: '',
```
Comment (albennett) [inline packages/stories/index.js]: The serviceId is found under the profile (channel). `channel.serviceId` 
Does it seem complex to get it?

Comment (anafilipadealmeida) [reply packages/stories/index.js]: Yes, it is a bit, as we will have to pass it through some components related to Posts. I've left a comment regarding it in the jira ticket: https://buffer.atlassian.net/browse/PUB-1972 and tagged you and Julian there

File: packages/stories/index.js [hunk 2]
```diff
@@ -61,6 +63,24 @@ export default connect(
     onPreviewClick: ({
       stories, profileId, id, scheduledAt,
     }) => {
+      const ctaProperties = getCtaProperties(SEGMENT_NAMES.STORIES_PREVIEW_QUEUE);
+      const imageCount = stories.filter(story => story.type === 'image').length;
+      const videoCount = stories.filter(story => story.type === 'video').length;
+      const noteCount = stories.filter(story => story.note && story.note.length > 0).length;
+      const metadata = {
+        product: 'publish',
```
Comment (albennett) [inline packages/stories/index.js]: I meant to say this in the segment call (I'll add it to the readme) but we don't need to pass in product here. I added product name as a window var and set it in the middleware so we don't need to pass it every time. https://github.com/bufferapp/buffer-publish/blob/master/packages/analytics-middleware/middleware.js#L17

File: packages/story-group-composer/index.js [hunk 3]
```diff
@@ -42,8 +45,26 @@ export default connect(
       dispatch(actions.handleSaveStoryNote({ note, order }));
     },
     onPreviewClick: ({
-      stories, profileId, id, scheduledAt,
+      stories, profileId, id, scheduledAt, serviceId,
     }) => {
+      const ctaProperties = getCtaProperties(SEGMENT_NAMES.STORIES_PREVIEW_COMPOSER);
+      const imageCount = stories.filter(story => story.type === 'image').length;
```
Comment (albennett) [inline packages/story-group-composer/index.js]: I think there might be other events that need the image/video/note counts. This is great for now but I might move it into the analytics-middleware utils if we end up needing to do this in several places.

File: packages/stories/index.js (hunk 2 above)
Comment (anafilipadealmeida) [reply packages/stories/index.js]: Good to know! It's probably worth doing some cleaning in the previous tracking, as it's being passed for instance in the plans package and it leads to a bit of confusion 🙂

File: packages/story-group-composer/index.js (hunk 3 above)
Comment (anafilipadealmeida) [reply packages/story-group-composer/index.js]: 🤔 I don't think the analytics utils is the best place for this to be honest. I've been thinking about some of the utils related to stories, and I believe they should be in a general utils package. Our structure is quite confusing at the moment, let me think a bit more about the utils 😅

File: packages/stories/index.js (hunk 2 above)
Comment (albennett) [reply packages/stories/index.js]: That's great to know, yes let's remove unneeded code 🔪

File: packages/stories/index.js (hunk 1 above)
Comment (albennett) [reply packages/stories/index.js]: That's great! Julian is out this week so Matt could be a good one to tag, too.

File: packages/story-group-composer/index.js [hunk 4]
```diff
@@ -42,8 +45,26 @@ export default connect(
...
+      const videoCount = stories.filter(story => story.type === 'video').length;
+      const noteCount = stories.filter(story => story.note && story.note.length > 0).length;
+      const metadata = {
```
Comment (albennett) [inline packages/story-group-composer/index.js]: We'll still want to pass `clientId` here but we'll want to pass null as value.

File: packages/story-group-composer/index.js [hunk 5]
```diff
@@ -42,8 +45,26 @@ export default connect(
...
+        channelId: profileId,
+        channelServiceId: serviceId,
+        mediaCount: imageCount + videoCount,
+        imageCount,
+        videoCount,
+        noteCount,
+        scheduledAt,
+        ...ctaProperties,
+      };
+      dispatch(analyticsActions.trackEvent('Story Preview Opened Composer', metadata));
```
Comment (albennett) [inline packages/story-group-composer/index.js]: This has now been changed to `Story Group Previewed` for eventName.

**Comment by albennett** [comment]
Hey @anafilipadealmeida 😄 is there currently anything blocking for this PR?

File: packages/story-group-composer/index.js (hunk 5 above)
Comment (anafilipadealmeida) [reply packages/story-group-composer/index.js]: Just to be clear @albennett , are the CTA names still : publish-stories-queue-preview-1 and publish-stories-composer-preview-1? Is the name now just `Story Group Previewed` for both the Queue and the Composer?

Comment (albennett) [reply packages/story-group-composer/index.js]: yep! They'll be able to filter the event by cta to learn where it's happening :)

**Pull request merged**